x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/plan/physical/ExchangeSinkExecSerializationTests.java

```diff
@@ -80,7 +80,7 @@ public void testManyTypeConflicts() throws IOException {
      * See {@link #testManyTypeConflicts(boolean, ByteSizeValue)} for more.
      */
     public void testManyTypeConflictsWithParent() throws IOException {
-        testManyTypeConflicts(true, ByteSizeValue.ofBytes(3271486));
+        testManyTypeConflicts(true, ByteSizeValue.ofBytes(3307704L));
```

Comment:
> Since FieldAttributes are being deduplicated but strings are not

I have a bit of a concern now that you mentioned this bit (thank you for explaining 👍 ): the test itself has a single level parent hierarchy for the fields it's creating. I am wondering if there will (or won't) be an exponential growth in serialized data size if the mapping would be many layers deep (I don't know... 4-5-6), something that would reflect better the [ECS mappings](https://github.com/elastic/ecs/blob/main/generated/csv/fields.csv).

Reply:
The concern on the advantage of field deduplication vs. fields only is reasonable.
For the record, I tested string caching and it gives us a 5-20% on these tests. I can revive it and submit a PR, maybe it can give advantages also for parent names.

Reply:
I'm working on adding a test to simulate a precarious situation with lots of nesting.

FWIW, I don't think the cost can be exponential, because the added cost is per fieldname that we serialize; shouldn't matter how nested the field hierarchy is, for every `nested.field.maybe.even.nested` we send an addtional string that's bounded by the field name, e.g. `nested.field.maybe.even` in this case.

Reply:
https://github.com/elastic/elasticsearch/pull/112929

Reply:
I added two tests. Both use a rather deep and wide mapping: 6 levels deep, each level has 9 children (per node), so it's 9^6 relevant fields.

For `FROM idx | LIMIT 10`, the plan size on main is ~100MB, with this PR it grows to 130MB.
For `FROM idx | LIMIT 10 | KEEP one_field` the plan size is ~20MB on both - it's absolutely dominated by the size of the serialized `EsIndex`, that's written as part of the `EsRelation`.

Reply:
> the plan size on main is ~100MB, with this PR it grows to 130MB

That's not insignificant. @nik9000 what's your gut feeling about this?